### DISCUSSION TITLE
Replace default loader during page load

### DISCUSSION THREAD

**Discussion by shrinidhin** [h2oai/wave#869] [Q&A] [answered]
Hi, Is there any way to replace the default Loading.. that appears on the page which is rendered when the wave app is run?I would like to show some coloured window pop-up or replace the loading with a customized message till my page is loaded, as I have some pre-processing to be done before that. I did try to use a progress bar but it doesn't seem to work for this scenario. Any inputs would be helpful.Thank you!

**Comment by vopani** [comment, marked as answer]
You can run your pre-processing as a [Background Task](https://wave.h2o.ai/docs/examples/background) and show a pop-up or card with any custom design / message in the meantime until the pre-processing is completed.

**Reply by shrinidhin** [reply]
It worked. Thanks!
